examples/upcloud-cli/main.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"git.sr.ht/~yoink00/goflenfig"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/client"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+	"github.com/davecgh/go-spew/spew"
+)
+
+var username string
+var password string
+
+func init() {
+	goflenfig.Prefix("UPCLOUD_")
+	goflenfig.StringVar(&username, "username", "", "UpCloud user name")
+	goflenfig.StringVar(&password, "password", "", "UpCloud password")
+
+	rand.Seed(time.Now().Unix())
+}
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	goflenfig.Parse()
+
+	command := flag.Arg(0)
+
+	if len(username) == 0 {
+		fmt.Fprintln(os.Stderr, "Username must be specified")
+		return 1
+	}
+
+	if len(password) == 0 {
+		fmt.Fprintln(os.Stderr, "Password must be specified")
+		return 2
+	}
+
+	fmt.Println("Creating new client")
+	c := client.New(username, password)
+	s := service.New(c)
+
+	switch command {
+	case "deleteservers":
+		if err := deleteServers(s); err != nil {
+			return 1
+		}
+	case "deletestorage":
+		if err := deleteStorage(s); err != nil {
+			return 2
+		}
+	case "createserver":
+		if err := createServer(s); err != nil {
+			return 3
+		}
+	default:
+		fmt.Fprintln(os.Stderr, "Unknown command: ", command)
+		return 99
+	}
+
+	return 0
+}
+
+func createServer(s *service.Service) error {
+	fmt.Println("Creating server")
+	details, err := s.CreateServer(&request.CreateServerRequest{
+		Hostname: "stuart.example.com",
+		Title:    fmt.Sprintf("example-cli-server-%04d", rand.Int31n(1000)),
+		Zone:     "fi-hel2",
+		Plan:     "1xCPU-1GB",
+		StorageDevices: []upcloud.CreateServerStorageDevice{
+			{
+				Action:  upcloud.CreateServerStorageDeviceActionClone,
+				Storage: "01000000-0000-4000-8000-000050010400",
+				Title:   "Centos8 from a template",
+				Size:    50,
+				Tier:    upcloud.StorageTierMaxIOPS,
+			},
+		},
+		IPAddresses: []request.CreateServerIPAddress{
```

Comment:
It would be better not to use IPAddresses field anymore in examples as it's functionality has been replaced by the Networking structure.

See https://developers.upcloud.com/1.3/8-servers/#create-server

Reply:
Hi Anton,

At the moment I am in the process of converting the 1.2 API from XML to JSON before moving on to using the 1.3 API. This example is correct as of 1.2? I will update it once I begin migrating from 1.2 to 1.3.

Thanks,

Stuart

Reply:
Hey Stuart,

(Apparently github ate my reply)

Anyway, the field is valid for 1.2. However, as it can't express the newer network features the usage is not encouraged.

Thanks,
Anton